libs/wingsdk/src/target-tf-aws/api.onrequest.inflight.ts

```diff
@@ -0,0 +1,76 @@
+import type { APIGatewayProxyEvent, APIGatewayProxyResult } from "aws-lambda";
+import type {
+  ApiRequest,
+  ApiResponse,
+  IApiEndpointHandlerClient,
+  Json,
+} from "../cloud/api";
+
+export class ApiOnRequestHandlerClient {
+  private readonly handler: IApiEndpointHandlerClient;
+  constructor({ handler }: { handler: IApiEndpointHandlerClient }) {
+    this.handler = handler;
+  }
+  public async handle(
+    request: APIGatewayProxyEvent
+  ): Promise<APIGatewayProxyResult> {
+    const apiRequest: ApiRequest = mapApigatewayEventToCloudApiRequest(request);
+    const apiResponse: ApiResponse = await this.handler.handle(apiRequest);
+    const apiGatewayResponse: APIGatewayProxyResult =
+      mapCloudApiResponseToApigatewayResponse(apiResponse);
+    return apiGatewayResponse;
+  }
+}
+
+/**
+ * Map a Cloud API response to an API Gateway response
+ * @param resp Cloud API response
+ * @returns API Gateway response
+ */
+function mapCloudApiResponseToApigatewayResponse(resp: ApiResponse) {
+  return {
+    statusCode: resp.status,
+    body: resp.body ? JSON.stringify(convertMapToObject(resp.body)) : "",
+    headers: convertMapToObject(resp.headers),
+  };
+}
+
+/**
+ * Converts the inflight return value to a JSON object
+ * @param map A map of key-value pairs
+ * @returns An object with the same key-value pairs
+ */
+function convertMapToObject(map: Json | undefined) {
+  if (!map) {
+    return undefined;
+  }
+  if (!(map instanceof Map)) {
+    // the inflight returns a Map instead of a plain object
+    throw new Error("Expected a Map");
+  }
+  return Object.fromEntries(map);
+}
```

Comment:
Sorry for the bother 😅 - I think we can remove this conversion since we have merged in a fix here: https://github.com/winglang/wing/pull/1653